### DISCUSSION TITLE
How to create multiple clients for wireguard mode

### DISCUSSION THREAD

**Discussion by phucnb** [mitmproxy/mitmproxy#6602] [Q&A]
How I can add more clients when using wireguard mode?
I know that I can add more clients by using --mode wireguard:client2.conf@51281 --mode wireguard:client3.conf@51282 and so on. But i have to stop the server and run the command again? Is there a way to add it without stop and run the command again or any way we can add it programmatically

**Comment by mhils** [comment]
> How I can add more clients when using wireguard mode?

Currently each WireGuard server is bound to one client, so you do need to specify `mode` multiple times. That limitation exists to simplify implementation. :)

You can update the `mode` option while mitmproxy is running (either via the CLI interface or via an addon), that will start new servers without shutting anything down.

**Reply by phucnb** [reply]
I found a workaround (not sure if this was a properly way). Basically I just change the address on the wg config file from 10.0.0.1 to something else like 10.0.0.2 then both of clients can connect
